solidity/test/rewards/TestECDSARewards.js

```diff
@@ -230,10 +230,130 @@ describe("ECDSARewards", () => {
         })
       }
 
-      // 2,376 * 2 = 4,752
       await assertKeepBalanceOfBeneficiaries(expectedBeneficiaryBalance.muln(2))
     })
 
+    it("should revert when rewards have been withdrawn already", async () => {
+      for (let i = 0; i < 8; i++) {
+        await keepFactory.stubOpenKeep(owner, operators, firstIntervalStart)
+      }
+
+      await timeJumpToEndOfIntervalIfApplicable(0)
+
+      // mark first keep as closed
+      const keepAddress0 = await keepFactory.getKeepAtIndex(0)
+      const keep0 = await BondedECDSAKeepStub.at(keepAddress0)
+      await keep0.publicMarkAsClosed()
+
+      await rewardsContract.receiveReward(keepAddress0)
+
+      for (let i = 0; i < operators.length; i++) {
+        await rewardsContract.withdrawRewards(0, operators[i], {
+          from: operators[i],
+        })
+      }
+
+      for (let i = 0; i < operators.length; i++) {
+        await expectRevert(
+          rewardsContract.withdrawRewards(0, operators[i], {
+            from: operators[i],
+          }),
+          "No rewards to withdraw"
+        )
+      }
+    })
```

Comment:
It's worth adding yet one test that will cover `receiveReward + withdrawRewards + receiveRewards + withdrawRewards` path for the same interval.

Reply:
I've added for: `receiveReward + withdrawRewards + receiveRewards` since the second `receiveRewards` will trigger revert: `Rewards already claimed`

Reply:
That's very good but what I suggested is yet another scenario. The scenario I have in mind is when you claim even more rewards by executing the second `receiveReward` for another keep. Real-world example is when that second keep gets closed later, once you have already received and withdrawn reward for the first keep.

Reply:
Okay, added in 09453fd